src/gate_detect/src/GateDetection.cpp

```diff
@@ -0,0 +1,330 @@
+            }
+        }
+    } else {
+        PoleDetection leftPole =
+        *std::min_element(vertPoles.begin(),
+                          vertPoles.end(),
+                          [](PoleDetection lhs, PoleDetection rhs) {
+                              return lhs.getVertMid() < rhs.getVertMid();
+                          });
+
+        PoleDetection rightPole =
+        *std::min_element(vertPoles.begin(),
+                          vertPoles.end(),
+                          [](PoleDetection lhs, PoleDetection rhs) {
+                              return lhs.getVertMid() > rhs.getVertMid();
+                          });
+
+        if (!horPoles.empty()) {
+            PoleDetection topPole =
+            *std::min_element(horPoles.begin(),
+                              horPoles.end(),
+                              [](PoleDetection lhs, PoleDetection rhs) {
+                                  return lhs.getHorMid() < rhs.getHorMid();
+                              });
+
+            if (leftPole.getVertMid() > topPole.getVertMid()) {
+                gateCoord[0] = 0;
+                gateCoord[1] = 0;
+                gateCoord[2] = 0;
+
+                gateCoord[3] = 1;
+                gateCoord[4] = rightPole.getVertAngle(_imagePixelWidth);
+                gateCoord[5] = rightPole.getVertDistance();
+
+                gateCoord[6] = 1;
+                gateCoord[7] = topPole.getHorAngle(_imagePixelHeight);
+                gateCoord[8] = topPole.getHorDistance();
+
+            } else if (rightPole.getVertMid() < topPole.getVertMid()) {
+                gateCoord[0] = 1;
+                gateCoord[1] = leftPole.getVertAngle(_imagePixelWidth);
+                gateCoord[2] = leftPole.getVertDistance();
+
+                gateCoord[3] = 0;
+                gateCoord[4] = 0;
+                gateCoord[5] = 0;
+
+                gateCoord[6] = 1;
+                gateCoord[7] = topPole.getHorAngle(_imagePixelHeight);
+                gateCoord[8] = topPole.getHorDistance();
+            } else {
+                gateCoord[0] = 1;
+                gateCoord[1] = leftPole.getVertAngle(_imagePixelWidth);
+                gateCoord[2] = leftPole.getVertDistance();
+
+                gateCoord[3] = 1;
+                gateCoord[4] = rightPole.getVertAngle(_imagePixelWidth);
+                gateCoord[5] = rightPole.getVertDistance();
+
+                gateCoord[6] = 1;
+                gateCoord[7] = topPole.getHorAngle(_imagePixelHeight);
+                gateCoord[8] = topPole.getHorDistance();
+            }
+
+        } else {
+            gateCoord[0] = 1;
+            gateCoord[1] = leftPole.getVertAngle(_imagePixelWidth);
+            gateCoord[2] = leftPole.getVertDistance();
+
+            gateCoord[3] = 1;
+            gateCoord[4] = rightPole.getVertAngle(_imagePixelWidth);
+            gateCoord[5] = rightPole.getVertDistance();
+        }
+    }
+    return gateCoord;
```

Comment:
This function is very long and seems somewhat repetitive. Is there any way we can shorten it by making a few functions?